Lab1 Implemented Fibonacci sequence

sirigithub opened the pull request:
Fixes #367 

Review of colleague's PR #

Changes proposed in this PR:

Commit:
Build Lab2 Bubble sort and bonus Insertion sort solutions and test cases

Commit:
feat(lab1): Create fibonacci number generator function

Added a function that takes an integer input and prints out the fibonacci sequence with that many iterations. This also includes the ability to print out negafibonacci.

Commit:
test(lab1): Test main function

Test written for the main function of fibonacci, lab1

Commit:
test(lab1): Test fib function

Add a series of tests to ensure the fibonacci function outputs correctly if given both positive and negative integers.

Commit:
refactor(lab1): Refactor fib function to use modulus

Based on a hint from Dan in the lecture today, I have refactored my implementation to use modulus and taken a more 'positive fibonacci first' approach. I have also added comments to help provide clarity on what each section of the code is doing.

Commit:
fix(typo): Fixed a typo in comment

Commit:
Implement 02_bubble

nicholascross committed:
Implement IPAddr type

- Implement IPAddr type backed by an array of 4 bytes
- Implement IPAddr related function to print IP address
- Add simple application to demonstrate formatting
- Add unit tests to cover string formatting of IPAddr

runnerdave committed:
Implement Bubble sort function plus tests

Commit:
Change test cases to be table driven and refine following PR review

runnerdave committed:
Implement feedback from PR review

Commit:
refactor(fib): Refactor fib function to use two variables instead of a map

Removed the map from fib function and replaced it with two integers to save memory.

Commit:
feat(fib): Add the ability to handle 0 as in input to fib

Supplying an input of 0 to fib will now print f0 (which is 0) in return.

Commit:
feat(fib): Add a check for integer overflow to fib function

Fib function now has a check to see if the input has caused a number overflow error and prints it out accordingly.
Added a few minor comment updates.

Commit:
test(fib): Add tests for fib function

Added a test for fib(0) and a test to check whether an out of bounds / number overflow error has ocurred.

Commit:
refactor(fib): Additional linting fix

Impatient me went and pushed the code before proof reading my comments. Additional commit to ensure they read correctly.

Commit:
Build Lab4 Numeronym solution and test cases

willshen24 committed:
-Completing Lab05 - Stringer with test cases
-Adding changes after PR reviews

pfannerj committed:
Build Lab5 Stringer solution and test cases

sirishavadrevu committed:
Implemented lab1 Fibonacci sequence

sirishavadrevu committed:
Merge branch 'master' into vadrevuslab1

pentaphobe committed:
Add @pentaphobe to CODEOWNERS

One more limb to the Voltron that is course reviewers

n0npax reviewed (dismissed):
lgmt

brnbke committed:
Lab3 adding letters and sortLetters Functions (#391)

sirigithub committed:
Merge branch 'master' into vadrevuslab1

juliaogris commented on 01_fib/sirigithub/main.go:
```diff
@@ -0,0 +1,37 @@
+		}
+		fmt.Fprintln(out, current*sign)
+		previous, current = current, (current + previous)
+	}
```
functionally correct. For completeness a slightly shorter version:
```go
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		fmt.Fprintln(out, n1)
	}
```

juliaogris commented on 01_fib/sirigithub/main.go:
```diff
@@ -0,0 +1,37 @@
+func main() {
+
+	fib(7)
+
```
ditto

juliaogris commented on 01_fib/sirigithub/main.go:
```diff
@@ -0,0 +1,37 @@
+}
+
+func main() {
+
```
remove unnecessary blank line

juliaogris reviewed (changes requested):
Please read through the [Collected Code Review Comments](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments)
- [Fix your pr description](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#pr-description)
- [Fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages)

Commit:
Stringer (#363)

* type IPAddr represents a IPv4 address and implements Stringer
  unspecified bytes are defaulted to 0

* main() prints the localhost IPv4 address followed by newline

saurabhvyas9 committed:
01_fib Lab1 commit for review - saurabhvyas9. (#225)

Implement 01_fib Lab1

Commit:
Implement Bubble/Insertion/Quick Sort (#242)

* New function bubble([]int) []int returns sorted copy
  of passed in slice using bubble sort
* New function insertion([]int) []int returns sorted copy
  of passed in slice using insertion sort
* New function quick([]int) []int returns sorted copy
  of passed in slice using quick sort
  Uses Lomuto partition scheme

sirigithub committed:
Merge branch 'master' into vadrevuslab1

sirishavadrevu committed:
Removed blank lines as per code review comments

sirishavadrevu committed:
Removed blank lines as per code review comments

sirishavadrevu committed:
Merge branch 'vadrevuslab1' of https://github.com/sirigithub/go-course into vadrevuslab1
